public/class-public.php

```diff
@@ -149,9 +163,53 @@ public function new_account_page_content() { ?>
     </div>
     <?php
   }
-}
+  public function customer_order_form_submit(){
+    if (isset($_POST['customer']) && $_POST['customer'] != null) {
+      setcookie("GIGFILLIATE_PLACING_ORDER_FOR_CUSTOMER", $_POST['customer'], time() + (86400 * 30), "/");
+    }
+    header('Location: ' . $_SERVER['HTTP_REFERER']);
+  }
+  public function exit_customer_form_submit(){
+    if (isset($_POST['exit_customer'])) {
+      setcookie("GIGFILLIATE_PLACING_ORDER_FOR_CUSTOMER",null, time() - 3600,"/");
+    }
+    header('Location: ' . $_SERVER['HTTP_REFERER']);
+  }
+  public function customer_notice(){
+    if(isset($_COOKIE["GIGFILLIATE_PLACING_ORDER_FOR_CUSTOMER"])) {
+      $customer = get_user_by("ID", $_COOKIE["GIGFILLIATE_PLACING_ORDER_FOR_CUSTOMER"]);
+      ?>
+      <div class="alert alert-info" role="alert">
+      You're placing an order for <?php echo $customer->first_name.' '.$customer->last_name;?>.
+      </div>
+      <?php
+    }
+  }
 
+  public function gofc_search_product(){
+    $args = array(
+      'post_type' => 'product',
+      'posts_per_page' => 15,
+      'order' => 'ASC',
+      's'=>$_GET["search"]
+    );
+    $to_return = [];
+    foreach ((new WP_Query( $args ))->posts as $post) {
+      $product = wc_get_product($post->ID);
+      $to_return[] = [
+        "id"=>$post->ID,
+        "thumbnail_url"=>wp_get_attachment_url($product->get_image_id()),
+        "name"=>$product->get_name(),
+        "price"=>$product->get_regular_price(),
+        "sku"=>$product->get_sku(),
+        "add_to_cart_url"=>$product->add_to_cart_url(),
+        "is_in_stock"=>$product->is_in_stock()
+      ];
+    }
+    echo json_encode($to_return);
+    wp_die();
```

Comment:
Instead of echo and wp_die use exit, example:
`exit( json_encode($to_return) );`